--- a/device/rockchip/common/scripts/mk-kernel.sh
+++ b/device/rockchip/common/scripts/mk-kernel.sh
@@ -81,6 +81,7 @@ do_build()
 			# The FIT image for initrd would be packed in rootfs stage
 			if [ -n "$RK_BOOT_FIT_ITS" ] && \
 				[ -z "$RK_ROOTFS_INITRD" ]; then
+				run_command cd "$RK_SDK_DIR"
 				run_command "$RK_SCRIPTS_DIR/mk-fitimage.sh" \
 					"kernel/$RK_BOOT_IMG" \
 					"$RK_BOOT_FIT_ITS" \